src/main/java/Duke.java

```diff
@@ -1,10 +1,204 @@
+import exceptions.*;
+import tasklist.Deadline;
+import tasklist.Event;
+import tasklist.Task;
+import tasklist.ToDo;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Scanner;
+
 public class Duke {
+    public static ArrayList<Task> taskList = new ArrayList<>(100);
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        UI.printDuke();
+        String line;
+        Scanner in = new Scanner(System.in);
+
+        while (true) {
+            line = in.nextLine();
+            if (line.equals("bye")) {
+                UI.bye();
+                break;
+            } else if (line.equals("list")) {
+                list();
+
+            } else if (line.contains("done") && line.charAt(0) == 'd') {
+                done(line);
+
+            } else if (line.contains("event") && line.charAt(0) == 'e') {
+                event(line);
+
+            } else if (line.contains("deadline") && line.charAt(0) == 'd') {
+                deadline(line);
+
+            } else if (line.contains("todo") && line.charAt(0) == 't') {
+                todo(line);
```

Comment:
I believe the line.contains(command) method cannot tell where the command begins in the user input string? So I was thinking if input string has "done" in it and the line.charAt(0) == 'd' is true, yet the first word can be e.g. "ddt"? The line logic seems not so clear?

Reply:
Thanks for the advice, will improvise the code accordingly!

Reply:
Thanks for the advice, will improvise the code accordingly!